dropboxfs.py

```diff
@@ -90,31 +83,120 @@ def close(self):
         if hasattr(self.temp, 'flush'):
             self.temp.flush()
         self.temp.seek(0)
-        self.client.put_file(self.name, self, overwrite=True)
+        self.client.files_upload(
+            self.temp.read(),
+            self.name,
+            mode=WriteMode.overwrite)
         self.temp.close()
 
 
-class SpooledReader(ContextManagerStream):
-    """
-    Reads the entire file from the remote server into a buffer or temporary
-    file. It can then satisfy read(), seek() and other calls using the local
-    file.
-    """
-    def __init__(self, client, name, max_buffer=MAX_BUFFER):
+class ChunkedReader(ContextManagerStream):
+    """ A file-like that provides access to a file with dropbox API"""
+    """Reads the file from the remote server as requested.
+    It can then satisfy read()."""
+    def __init__(self, client, name):
         self.client = client
-        r = self.client.get_file(name)
-        self.bytes = int(r.getheader('Content-Length'))
-        if r > max_buffer:
-            temp = tempfile.TemporaryFile()
-        else:
-            temp = StringIO()
-        shutil.copyfileobj(r, temp)
-        temp.seek(0)
-        super(SpooledReader, self).__init__(temp, name)
+        try:
+            _, response = self.client.files_download(name)
+            self.r = response.raw
```

Comment:
This converts the `requests.Response` object into a `urllib3.response.HTTPResponse` object. So, it shouldn't. It's just a file-like object then.